### PR TITLE
fix: [OIP-579]: Update to find selected health source via health source identifier

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/health-source/common/utils/HealthSource.utils.ts [hunk 1]
```diff
@@ -284,3 +285,9 @@ export const getConnectorRef = (connectorRef?: string | { value: string }): stri
 
   return connectorRef?.value
 }
+
+export const getSelectedHealthSource = (sourceData: SourceDataInterface): UpdatedHealthSource => {
+  return sourceData?.healthSourceList?.find(
+    source => source.identifier === sourceData.healthSourceIdentifier
+  ) as UpdatedHealthSource
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/common/utils/HealthSource.utils.ts:292]: forcing UpdatedHealthSource might be wrong , as .find() can return undefined value unmatched  value

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.utils.ts [hunk 2]
```diff
@@ -81,9 +81,7 @@ export const convertStringMetricPathToObject = (metricPath: string | MetricPathD
 }
 
 export const createAppDynamicsData = (sourceData: any): AppDynamicsData => {
-  const payload: UpdatedHealthSource = sourceData?.healthSourceList?.find(
-    (source: UpdatedHealthSource) => source.identifier === sourceData.healthSourceIdentifier
-  )
+  const payload: UpdatedHealthSource = getSelectedHealthSource(sourceData)
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.utils.ts:84]: don't need `payload: UpdatedHealthSource`. as return type of getSelectedHealthSource is defined

File: src/modules/85-cv/pages/health-source/common/utils/HealthSource.utils.ts (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/common/utils/HealthSource.utils.ts]: Done

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.utils.ts (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.utils.ts]: Done

**Review by deepesh-ui — APPROVED** [review]

**Review by karansaraswat19 — APPROVED** [review]

**Pull request merged**